oojs.js

```diff
@@ -3,10 +3,28 @@
   // A songs property that is determined by some input. It should contain multiple songs, each of which can just be a song title.
   // A favorites property that is initialized at 0. This cannot be set by user input.
   // An addSong method that adds a song to the songs property.
+  class Playlist {
+  	constructor(title, songs_array ) {
+  		this.title = title,
+  		this.songs = songs_array,
```

Comment:
`new Playlist()` might be called without a value for `songs_array`, to make sure `songs` is initialized to an array try this:
```
// if `songs_array` is undefined, we'll set it to an empty array
this.songs = songs_array || []
```